Danki2/Assets/Scripts/Abilities/Ability.cs

```diff
@@ -1,9 +1,35 @@
-﻿namespace Abilities
+﻿using Assets.Scripts;
+
+namespace Abilities
 {
-    public enum Ability
+    public abstract class AbilityContext
     {
-        SLASH,
-        SHIELD_BASH,
-        OTHER
+        private readonly AbilityCaster owner;
+        private readonly AbilityCaster target;
+        private readonly float xOrigin;
+        private readonly float yOrigin;
```

Comment:
We would also want a Z component, and would be nice just as 1 Vector3.

Reply:
done.